File: ios/MobilePassSDK/Model/PassFlowResult.swift
//
//  PassFlowResult.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 1.06.2023.
//

import Foundation

public struct PassFlowResult {
    public var result:      Int
    public var direction:   Int?
    public var clubId:      String?
    public var clubName:    String?
    public var states:      [PassFlowState]
    
    init(result: PassFlowResultCode, states: [PassFlowState], direction: Direction?, clubId: String?, clubName: String?) {
        self.result     = result.rawValue
        self.direction  = direction != nil ? direction!.rawValue : nil
        self.clubId     = clubId
        self.clubName   = clubName
        self.states     = states
    }
}

File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/PassFlowResult.java
package com.armongate.mobilepasssdk.model;

import java.util.List;

public class PassFlowResult {
    public Integer  result;
    public Integer  direction;
    public String   clubId;
    public String   clubName;
    public List<PassFlowState> states;
    public String   message;
    public String   code;

    public PassFlowResult(Integer result, Integer direction, String clubId, String clubName, List<PassFlowState> states, String message, String code) {
        this.result     = result;
        this.direction  = direction;
        this.clubId     = clubId;
        this.clubName   = clubName;
        this.states     = states;
        this.message    = message;
        this.code       = code;
    }
}
